openstack/networking/v2/extensions/quotas/results.go

```diff
@@ -56,3 +75,50 @@ type Quota struct {
 	// SubnetPool represents a number of subnet pools. A "-1" value means no limit.
 	SubnetPool int `json:"subnetpool"`
 }
+
+// QuotaDetailSet represents details of both operational limits of Networking resources for a project
+// and the current usage of those resources.
+type QuotaDetailSet struct {
```

Comment:
This set is potentially problematic, because it's generated dynamically. It can certainly differ between clouds. E.g. my cloud outputs a 'trunk' resource:

```
{
  "quota": {
    "network": {
      "limit": -1,
      "used": 2,
      "reserved": 0
    },
    "subnet": {
      "limit": -1,
      "used": 2,
      "reserved": 0
    },
    "subnetpool": {
      "limit": -1,
      "used": 0,
      "reserved": 0
    },
    "port": {
      "limit": -1,
      "used": 14,
      "reserved": 0
    },
    "router": {
      "limit": -1,
      "used": 2,
      "reserved": 0
    },
    "floatingip": {
      "limit": -1,
      "used": 2,
      "reserved": 0
    },
    "rbac_policy": {
      "limit": -1,
      "used": 0,
      "reserved": 0
    },
    "security_group": {
      "limit": -1,
      "used": 4,
      "reserved": 0
    },
    "security_group_rule": {
      "limit": -1,
      "used": 46,
      "reserved": 0
    },
    "trunk": {
      "limit": -1,
      "used": 6,
      "reserved": 0
    }
  }
}
```

'trunk' originates here:

https://opendev.org/openstack/neutron-lib/src/branch/master/neutron_lib/api/definitions/trunk.py#L49

It is registered as a quota resource in neutron here:

https://opendev.org/openstack/neutron/src/branch/master/neutron/extensions/trunk.py

I did a quick audit of other resources which are registered in the same way and only found 'router', 'floatingip', and 'subnetpool', all of which you already have.

At a minimum you should add 'trunk'. It would again be down to the style goals of gophercloud whether this data should be returned as a struct or a map.

Something else to call out: if we continue to return a struct, any field in this struct which wasn't returned by the server will be filled with {InUse: 0, Reserved: 0, Limit: 0}. This is valid quota, so the caller wouldn't be able to distinguish this from the resource not being present at all. This might be ok, but we should probably make this decision deliberately.

Reply:
[...]
> At a minimum you should add 'trunk'. It would again be down to the style goals of gophercloud whether this data should be returned as a struct or a map.

I'll add `trunk` (since we'll need it for our needs), but I suspect there are more extensions that register a resource in the quotas, they can be added later.

Reply:
Every field in this struct registered dynamically, btw, not just trunk. router, floatingip, and subnetpool are also registered in the same way. Also, the whole of neutron seems to be extensions. I wouldn't call out trunk as being different: it's probably just the most recent extension to be added. With an architecture like this you can almost guarantee that the api docs for quota detail will often/usually be out of date.

Reply:
What is the expected QuotaDetail value when the resource is not in the response?

If this question has a satisfying answer, then the struct would be much more usable than a map on the consumer side.